# Week 1

## Biological background

Bioinformatics deals with the analysis of biological sequences.
These sequences originate from organic macromolecules that play important roles in cells.
In this section, we describe these sequences and the biological processes involved in generating and maintaining them.
This section provides important background material of the course.

---

### Proteins

Proteins are large, complex macromolecules that play many important roles in the body.
They are critical to most of the work done by cells and are required for the structure, function and regulation of the body’s tissues and organs.
The basic building blocks of proteins are amino acids.

---

#### Amino acids

An amino acid contains a central carbon atom (called α-carbon, or C<sub>α</sub>) ({numref}`aminoacid`).
The α-carbon is bound to an amino group (NH<sub>2</sub>), a carboxyl group (COOH), and a hydrogen atom. In addition, each amino acid has a specific residue (R) group.

:::{figure} images/Week1/aminoacid.jpg
:alt: Structure of an amino acid
:width: 100%
:name: aminoacid

The structure of an amino acid. Four groups are connected to the α-carbon:
an amino group, a hydrogen atom, a carboxyl group, and a side chain (R group).
Credits: Rice University, CC BY 4.0 https://creativecommons.org/licenses/by/4.0, via OpenStax.
:::

:::{important}
Amino acids differ in their chemical properties, which are determined by their R groups.
It is important to know the amino acids, their one-letter and three-letter abbreviation, and their fundamental properties as given in the table.
:::

```{list-table}
:header-rows: 1
:widths: auto
:name: aminoacidtable

* - Amino acid
  - One-letter code
  - Three-letter code
  - Property
* - Arginine
  - Arg
  - R
  - Positively charged
* - Histidine
  - His
  - H
  - Positively charged
* - Lysine
  - Lys
  - K
  - Positively charged
* - Aspartic acid
  - Asp
  - D
  - Negatively charged
* - Glutamic acid
  - Glu
  - E
  - Negatively charged
* - Serine
  - Ser
  - S
  - Polar uncharged
* - Threonine
  - Thr
  - T
  - Polar uncharged
* - Asparagine
  - Asn
  - N
  - Polar uncharged
* - Glutamine
  - Gln
  - Q
  - Polar uncharged
* - Alanine
  - Ala
  - A
  - Hydrophobic
* - Valine
  - Val
  - V
  - Hydrophobic
* - Isoleucine
  - Ile
  - I
  - Hydrophobic
* - Leucine
  - Leu
  - L
  - Hydrophobic
* - Methionine
  - Met
  - M
  - Hydrophobic
* - Phenylalanine
  - Phe
  - F
  - Hydrophobic and aromatic
* - Tyrosine
  - Tyr
  - Y
  - Hydrophobic and aromatic
* - Trypotophan
  - Trp
  - W
  - Hydrophobic and aromatic
* - Glycine
  - Gly
  - G
  - Special (only H as side chain)
* - Proline
  - Pro
  - P
  - Special (side chain bound to backbone nitrogen)
* - Cysteine
  - Cys
  - C
  - Special (forms disulfide bonds)
```

Some amino acids have non-polar side chains, and these are generally **hydrophobic**, i.e., they do not like water.
Thus, they can often be found in the interior of proteins together with other hydrophobic amino acids.
**Aromatic** amino acids contain aromatic rings, and often stabilize folded protein structures.

In contrast, the charged and the polar amino acids are **hydrophilic**, i.e., they like water and can interact with it.
They can often be found on the surface of proteins or in the interior, when they can interact with another oppositely charged amino acid.
**Positively charged** amino acids, are also called basic amino acids and **negatively charged** amino acids are also called acidic amino acids.

Although amino acids can be classified into these groups based on their properties, some amino acids stand out.
The smallest amino acid is glycine, which provides great flexibility due to its small size.
In contrast, proline is an amino acid, where the side chain is bonded to the backbone nitrogen atom, which makes it very rigid.  

---

#### Protein structure

A protein is made up of one or more long, folded chains of amino acids (each called a **polypeptide**).
The 3D structure of a protein is also called its **conformation**.
The protein conformation is described on four levels - primary to quaternary structure ({numref}`struclevels`).

:::{figure} images/Week1/struclevels.jpg
:alt: The four levels of protein structure
:width: 100%
:name: struclevels

The four levels of protein structure.
Credits: Rao, A. Ryan, K. and Tag, A. Department of Biology, Texas A&M University.
:::

The structure of a protein is critical for its function.
For example, in an enzyme, the active site must be in the correct structure to be able to bind the substrate.
Other proteins might bind proteins (and influence their activity) or bind DNA (and regulate gene expression).
Additionally, some proteins are secreted from the cell or might function within the cell membrane.
Finally, proteins are often modified after protein synthesis (see translation), called post-translational modification.
These modifications can be important for protein function.

---

##### Primary structure

In a protein, amino acids are connected by covalent bonds, called peptide bonds.
A peptide bond connects one amino acid's carobxyl group and the next amino acid's amino group ({numref}`peptidebond`).
The sequence of amino acids linked by peptide bonds is called the **primary structure**.
The protein sequence is determined by the gene sequence encoding the protein.
The continuous chain of atoms along the protein is also called the **backbone**, it consists of the three backbone atoms (nitrogen, C<sub>α</sub>, carbon).

:::{figure} images/Week1/peptidebond.jpg
:alt: Peptide bond
:width: 100%
:name: peptidebond

A peptide bond connecting two amino acids.
Credits: Rice University, CC BY 4.0 https://creativecommons.org/licenses/by/4.0, via OpenStax.
:::

Each protein has a free amino group on one end, called the **N terminus**.
The other end has a free carboxyl group, called the **C terminus**.

:::{note}
As there are 20 distinct amino acids, there can be a huge number of different polypeptide chains, i.e., 20<sup>n</sup> for a polypeptide of length n.
Most of these potential sequences do not adopt a stable conformation, thus only a tiny fraction of these possibilities exist in nature.
:::

---

##### Secondary structure

Secondary structures are local conformations in the protein that are stabilized by hydrogen bonds between backbone atoms.
We distinguish the regular helices (i.e., α-helix) and sheet structures (i.e., β-sheet) ({numref}`secstructure`) and irregular turns.

**α-helices** are stabilized by hydrogen bonds between the oxygen atom in the C group in one amino acid and the hydrogen in the N group of the amino acids that is four amino acids farther along the chain.
Every helical turn has 3.6 amino acids residues and the side chains stick out of the helix.

β-pleated sheets (short: **β-sheets**) consist of β-strands, where the R groups extend above and below the strands.
The strands have a direction determined by the N- and C-terminus of the protein and are usually depicted as an arrow pointing towards the C-terminus.
Depending on the direction, strands can align parallel or antiparallel to each other.

:::{figure} images/Week1/secstructure.jpg
:alt: Secondary structure elements
:width: 100%
:name: secstructure

α-helices and β-sheets are stablized by hydrogen bonds between the backbone of proteins, i.e., the side chains are not involved.
The hydrogen bonds form between the oxygen atom in the C group in one amino acid and the hydrogen in the N group.
Black = carbon, White = hydrogen, Blue = nitrogen, and Red = oxygen. Credit:
Credits: Rao, A., Tag, A. Ryan, K. and Fletcher, S. Department of Biology, Texas A&M University.
:::

**Turns** are short secondary structure elements that are stabilized by hydrogen bonds between amino acids that are 1 to 5 peptide bonds away.
The most common form are β-turns, which connect antiparallel β-strands.

:::{note}
Although secondary structure elements are formed by hydrogen bonds between the backbone, certain amino acids are favoured in secondary structures and others are disfavoured.
For example, methionine, alanine, leucine, and glutamic acid are favoured in α-helices, whereas proline, glycine, and tyrosine are disfavoured.
Also, valine, isoleucine, tyrosine, cysteine, tryptophan, phenylalanine, and threonine are more frequently found in β-sheets, compared to α-helices.
In turns, glycine, asparagine, proline, and serine are preferred.
These preferences are used to predict secondary structure elements in proteins (see Week 4).
:::

The peptide bond is very rigid and planar, i.e., it cannot rotate to form the elements of protein structure.
However, the N-C<sub>α</sub> and the C<sub>α</sub>-C bonds can freely rotate, being only limited by the size and properties of the R-groups.
The 3D shape of the polypeptide backbone is thus determined by two **torsion angles**:
phi (φ) between N and C<sub>α</sub> and psi (ψ) between C<sub>α</sub> and C ({numref}`phipsi`A).
Although φ and ψ can rotate in principle, steric hindrance prevents certain combinations of angles, i.e., the bulkiness of the R-groups restricts the possible conformations.
Thus, certain combinations of φ and ψ are preferred.
We can plot the combinations of φ and ψ in a protein, called the **Ramachandran plot** ({numref}`phipsi`B).

The regular secondary structure elements (α-helix and β-sheet) contain consecutive amino acids with similar (φ,ψ) values.
These regions are typically highly populated in a Ramachandran plot.
Thus, the Ramachandran plot can be used to assess how plausible a predicted protein structure is.

:::{figure} images/Week1/phipsi.jpg
:alt: Phi, psi, and Ramachandran plot
:width: 100%
:name: phipsi

A) Peptide bond, φ, and ψ.
B) A typical Ramachandran plot. The regions marked "core" do not have any steric hindrance.
Yellow areas are generally allowed.
White areas represent conformations that are generally sterically unfavorable.
Credits: Choudhuri (2014). Bioinformatics for Beginners. doi: 10.1016/B978-0-12-410471-6.00008-6.
:::

:::{seealso}
You can find an illustrative animation on φ and ψ [here](https://www.youtube.com/watch?v=Q1ftYq13XKk).
:::

---

##### Tertiary structure

In contrast to the secondary structure, the tertiary structure of a protein involves interactions between the side chains ({numref}`terstructure`).
Thus, the chemical properties of the amino acids are very important for the tertiary structure. Different types of interactions stabilize the tertiary structure:

* Hydrogen bonds involving polar amino acids,
* ionic bonds between positively and negatively charged amino acids,
* hydrophobic R groups that tend to lie in the protein's interior, stabilized by hydrophobic interactions, and
* disulfide bonds (i.e., covalent bonds between cysteines).

:::{figure} images/Week1/terstructure.jpg
:alt: Tertiary structure interactions
:width: 100%
:name: terstructure

Chemical interactions that stabilize the tertiary structure of proteins.
Credits: Rice University, CC BY 4.0 https://creativecommons.org/licenses/by/4.0, via OpenStax.
:::


:::{admonition} Denaturation
:class: note
The noncovalent bonds that stabilize the protein structure are broken at high temperature.
Thus, most proteins unfold above about 60°C.
This process is called denaturation and is generally irreversible.
When proteins denature, they loose their function.
:::

**Domains** are distinct functional and/or structural units in a protein and are typically 50 to 350 amino acids long.
Usually, a domain is responsible for a particular function or interaction, contributing to the overall role of a protein.
A domain can exist in different contexts with other domains ({numref}`domains`).
In a multidomain protein, each domain folds independently of the others.

:::{figure} images/Week1/domains.jpg
:alt: Domain examples
:width: 100%
:name: domains

A) Example of a Src homology 3 (SH3) domain that is involved in protein-protein interaction. SH3 domains occur in a diverse range of proteins with different functions.
B) The cytoplasmic protein Nck contains multiple SH3 domains.
C) Domain composition of phospholipase D1, which has multiple functional domains that contribute to its overall function.
Credits: EMBL-EBI, CC BY 4.0 https://creativecommons.org/licenses/by/4.0.
:::

[comment]: # (domain examples taken from https://www.ebi.ac.uk/training/online/courses/protein-classification-intro-ebi-resources/protein-classification/what-are-protein-domains/)

---

##### Quaternary structure

Finally, individual folded polypeptides can interact to form **protein complexes**, also called quaternary structures.
Quaternary structure is stabilized by the same types of interactions as the tertiary structure.
The difference is that the amino acids involved belong to different polypeptides.

Many functional proteins are composed of multiple subunits, they are also called **oligomers** ({numref}`oligomers`).
The subunits can originate from the same protein sequence (called a homomer) or from different sequences (called a heteromer).
Proteins consisting of two subunits are also called dimer.

:::{figure} images/Week1/oligomers.jpg
:alt: Examples of oligomers
:width: 100%
:name: oligomers

Examples of oligomers.
A) Myoglobin, a heteromer of four subunits (PDB structure 1HV4 colored by chain).
B) UDP-galactose 4-epimerase, a homodimer (PDB structure 1EK5 colored by secondary structure).
:::

---

##### Substitutions

Mutations in the gene sequence can lead to changes in the primary structure of the protein, e.g., a substitution of one amino acid by a different one.
Often, such substitutions still lead to highly similar protein structures that perform the similar or even the same function, especially when the exchanged amino acids have similar chemical properties.
Nevertheless, single amino acid substitutions can have severe consequences.
A prominent example is sickle cell anemia, where a substitution of valine to glutamic acid in hemoglobin β results in a structural change that leads to a distortion in red blood cells ({numref}`sicklecell`).

:::{figure} images/Week1/sicklecell.jpg
:alt: Sickle cell anemia
:width: 100%
:name: sicklecell

Consequences of a substitution in  hemoglobin β resulting in sickle cell anemia.
Credits: Rao, A., Tag, A. Ryan, K. and Fletcher, S. Department of Biology, Texas A&M University.
:::

---

##### Visualization

There are many styles to view molecular structures.
You can see some examples in {numref}`protrep`.

:::{figure} images/Week1/protrep.jpg
:alt: Protein representation
:width: 100%
:name: protrep

Different representations of the PDB structure 5PEP generated with NGL.
:::

:::{seealso}
Most of the figures in this section are taken from [OpenStax](https://openstax.org/books/biology-2e/pages/3-4-proteins), where you can also find more information on proteins.
:::

---

### The flow of genetic information

#### Nucleic acids

Deoxyribonucleic acid (DNA) carries the genetic information of organisms and ribonucleic acid (RNA) is involved in the protein expression and is also the genetic material of some viruses.
Thus, these molecules are highly important as the basis of life on Earth.
The **genome** denotes the cell's entire genetic content and **genomics** is the study of genomes.

DNA and RNA are comprised of monomers called **nucleotides**, which are comprised of three components ({numref}`nucleotide`):
* A **pentose** sugar, where carbon residues are numbered 1' to 5' (read 1' as "one prime"). The type of pentose distinguishes RNA and DNA: the sugar is deoxyribose in DNA and ribose in RNA.
They are similar in structure, but deoxyribose has an H instead of an OH at the 2′ position.
* A **phosphate** group that is attached to the 5' position of the sugar.
* A **base** that is attached to the 1' position of the sugar.

:::{figure} images/Week1/nucleotide.jpg
:alt: Nucleotide overview
:width: 100%
:name: nucleotide

The components of a nucleotide.
Credits: Rice University, CC BY 4.0 https://creativecommons.org/licenses/by/4.0, via OpenStax.
:::

The bases can be divided into two categories: purines (with a double ring structure) and pyrimidines (with a single ring structure) ({numref}`nucleotide`). DNA contains A, T, C, and G; whereas RNA contains A, U, C, and G.

:::{important}
Nucleotides are central molecules of life. You do not need to remember the exact chemical structure, but you need to know the difference between DNA and RNA, the different bases and their category (purines or pyrimidines).
:::

---

#### The DNA double helix

The DNA molecule is a polymer of deoxyribonucleotides and forms a right-handed double helix.
The sugar and phosphate are on the outside forming the helix's backbone and the bases are stacked in the interior and bind each other by hydrogen bonds.
Thereby A pairs with T via two hydrogen bonds and C pairs with G via three hydrogen bonds, they are **complementary** bases.
These pairings are also called Watson-Crick base-pairing, named after the discoverers of DNA.

:::{figure} images/Week1/dna.jpg
:alt: DNA structure
:width: 100%
:name: dna

The DNA structure.
Credits: Nature Education.
:::

[comment]: # (figure sources
https://openstax.org/books/biology-2e/pages/3-5-nucleic-acids
https://www-nature-com.ezproxy.library.wur.nl/scitable/topicpage/discovery-of-dna-structure-and-function-watson-397/)

The two strands of the helix run in opposite directions, also called anti-parallel, i.e., one goes from 5' to 3' and the other from 3' to 5' ({numref}`nucleotide`).
The nucleotide sequence is typically written in 5' to 3' direction.
Due to the complementarity, the base sequence of a strand can be deduced from the base sequence from the other strand.
This is called the **reverse complement**.
For example, the reverse complement of AAGT is ACTT, where both strands are given in 5' to 3' direction.

---

#### DNA replication

As the two DNA strands are only connected via hydrogen bonds, they can be separated relatively easily, for example during DNA replication.
The separated strands each serve as a template on which a new complementary strand is synthesized by the enzyme DNA polymerase in 5' to 3' direction.
This mode of replication is called semiconservative .

:::{figure} images/Week1/replication.jpg
:alt: Replication
:width: 100%
:name: replication

A) The process of DNA replication.
Credits: Medprime, CC BY SA 1.0 https://creativecommons.org/publicdomain/zero/1.0/deed.en, via Wikimedia.
B) Semiconservative DNA replication, where the two copies each contain on original strand and one new strand.
Credits: modified from Rgranit, CC BY SA 2.5 https://creativecommons.org/licenses/by-sa/2.5/deed.en, via Wikimedia.
:::

The error rate of DNA replication is remarkably low, about one erroneous base in 10<sup>9</sup> bases.
This property preserves the genetic information during cell division, and also over generations.

---

#### RNA

In contrast to DNA, RNA does not form a stable double helix.
RNA is mainly single stranded, but most RNAs show intramolecular base pairing between complementary bases.
There are four major types of RNA: messenger RNA (mRNA), ribosomal RNA (rRNA), transfer RNA (tRNA), and microRNA (miRNA).

---

#### The central dogma of molecular biology

According to the central dogma of molecular biology, the flow of genetic information is essentially in one direction: from DNA via RNA to proteins ({numref}`dogma`).
Nevertheless, there are also genes that do not code for proteins, but where functional RNA is the end product. Furthermore, mobile genetic elements and viruses can encode reverse transcriptases (which can synthesize DNA from an RNA template) or RNA dependent RNA polymerases (which can replicate RNA).

:::{figure} images/Week1/dogma.jpg
:alt: Central dogma
:width: 100%
:name: dogma

The centra dogma of molecular biology.
Credits: Dhorspool, CC BY SA 3.0 https://creativecommons.org/licenses/by-sa/3.0/deed.en, via Wikimedia.
:::

During **transcription**, RNA polymerase reads the template strand (also called noncoding strand) in the 3' to 5' direction ({numref}`transcription`). This produces an RNA molecule from 5' to 3', a copy of the coding strand (where thymine is replaced by uracil).

:::{figure} images/Week1/transcription.jpg
:alt: Transcription
:width: 100%
:name: transcription

Transcription.
Credits: public domain, via Wikimedia.
:::

During protein **translation**, ribosomes synthesize polypeptides from messenger RNA (mRNA) ({numref}`translation`).
Thereby tRNAs decode the information on the RNA into amino acids, where a codon consisting of three nucleotides encodes the information for one amino acid.

:::{figure} images/Week1/translation.jpg
:alt: Translation
:width: 100%
:name: translation

Overview of translation.
:::

:::{seealso}
The details of transcription and translation differ between prokaryotes and eukaryotes. You can look up Chapters 15 and 16 of [Biology 2e](https://openstax.org/details/books/biology-2e) to learn more.
:::

---

#### The genetic code

The **genetic code** shows the correspondence between codons and amino acids ({numref}`geneticcode`).
Since 64 possible codons code for 20 different amino acids, the genetic code is degenerate, i.e., most amino acids are specified by more than one codon.
Thus, the protein sequence can be deduced from the DNA or RNA sequence but not vice versa.
There are three codons that do not encode for an amino acid, but instead signal the end of the protein sequence, called **stop codons**.
Furthermore, translation generally starts with the start codon AUG encoding methionine.
More information of how protein information is encoded in genomes can be found in the section on [genome annotation](sec_annotation).

:::{figure} images/Week1/geneticcode.jpg
:alt: The genetic code
:width: 100%
:name: geneticcode

The universal genetic code. Note that exceptions to this code exist, for example the vertebrate mitochondrial code.
Credits: SrhGreenwood, CC BY 4.0 https://creativecommons.org/licenses/by/4.0, via Wikimedia.
:::

:::{important}
The universal genetic code is very important to understand how information flows from genes to proteins. Nevertheless, you do not need to recall it, but can always look it up. When needed, it will also be provided in the exam.
:::

---

(sec_annotation)=
## Genome annotation

Anne: I suggest to mention this first, since it links well with the previous sections. So far, transcription is not well described in the background.

---

## Databases

### Introduction

Databases are at the core of bioinformatics.
In all analyses, we integrate pre-existing data and we need to access this data.
The journal Nucleic Acids Research publishes an entire issue in the beginning of each year on new and updated databases.
The list of these databases can also be accessed [online](https://www.oxfordjournals.org/nar/database/c).

Computer scientists have developed different kinds of databases.
One example are relational databases, which can be queried by **SQL** (structured query language) and which perform well for data that is processed computationally.
Another example are **XML** (extended markup language) databases which store data in specified well-structured XML files.
Nevertheless, most databases for biological sequence data use **flat file databases**, where the data is saved in structured text files.
This data can be manipulated in a text editor without requiring an additional program for database management, and they can be easily exchanged between scientists.
On the downside, searching them has a lower performance.
This is why they are often indexed, i.e., they contain an **index** of keywords, similar to a glossary in a book.

Depending on the kind of data included, we distinguish different kinds of biological databases:
* **Primary databases** contain primary sequence information from experimentally derived data that is directly submitted by the scientists that generated the data.
* **Secondary databases** provide the results of analyses of the information in primary databases.

Each entry in a database has a unique **accession number**.
This number is permanent and provides an unambiguous way to link to the entry.
The information that the accession refers to should not change.
To still allow updates to an entry, the accession number can contain a **version**, usually after a dot.
For example, NC_003070.9 is the latest version for Arabidopsis thaliana chromosome 1 in RefSeq.

Database entries often link to each other via **cross links**.

**TODO** mention flat/binary files, the knowledge clip could also be included.

---

### GenBank

[GenBank](https://www.ncbi.nlm.nih.gov/genbank/) is a popular primary database for nucleotide sequences and is based at the [NCBI](https://www.ncbi.nlm.nih.gov/) (National Center for Biotechnology Information).
A GenBank release occurs every two months and the most recent [release](https://www.ncbi.nlm.nih.gov/genbank/release/current/) from 15 June 2023 contains ~240 million sequences and additionally ~3.4 billion WGS (whole genome shotgun) records.
The latter are genome assemblies or genomes that were not yet completed.
The complete database is available for download via FTP, but the most convenient way to access individual entries is via the search on the GenBank website ({numref}`genbank`).

:::{figure} images/Week1/genbank.png
:alt: Genbank website
:width: 100%
:name: genbank

A screenshot of the GenBank website.
:::

:::{note}
These days it is required for publication that scientists submit their sequence data to GenBank or an associated database.
:::

Since data is directly submitted to GenBank, the information for some loci can be highly redundant.
The sequence records are owned by the original submitter and cannot be altered by someone else.

Genbank is part of the [INSDC](https://www.insdc.org/) (International Nucleotide Sequence Database Collaboration).
The other two member databases are [ENA](https://www.ebi.ac.uk/ena/browser/home) (European Nucleotide ARchive) and [DDBJ](https://www.ddbj.nig.ac.jp/index-e.html) (DNA Data Bank of Japan).
The data submitted to either database is exchanged daily, so all databases contain essentially the same information.

The **GenBank file format** is also a popular format to represent genes or genomes.
[Here](https://www.ncbi.nlm.nih.gov/Sitemap/samplerecord.html) you can find an example GenBank record with annotations.
Important elements are the Locus, Definition (i.e., the name), and the Organism.
Additionally, Features, such as genes and CDSs (coding sequences) are listed.

---

### RefSeq

The Reference Sequence ([RefSeq](https://www.ncbi.nlm.nih.gov/refseq/)) collection is also hosted at NCBI and contains genomic DNA, transcripts, and proteins.
The aim of RefSeq is to provide non-redundant, curated data.
RefSeq genomes are copies of selected assembled genomes in GenBank. Additionally, transcript and protein records are generated by several processes:

* Computation via the [eukaryotic](https://www.ncbi.nlm.nih.gov/genome/annotation_euk/) or [prokaryotic](https://www.ncbi.nlm.nih.gov/genome/annotation_prok/) annotation pipeline
* manual curation
* transfer of information from annotated genomes in GenBank

In contrast to GenBank, RefSeq records are owned by NCBI and can be updated to maintain annotation.
The current release is 218 from 5 May 2023 and contains ~261 million proteins from ~134,000 organisms.

The RefSeq accessions directly provide information on [molecule types](https://www.ncbi.nlm.nih.gov/books/NBK21091/table/ch18.T.refseq_accession_numbers_and_mole/?report=objectonly).
For example, `NC_` accessions denote complete genomes, `NP_` accessions denote proteins in one genome, and `WP_` accessions denote proteins in multiple genomes.

[comment]: # (sources:
  https://www.ncbi.nlm.nih.gov/books/NBK50679/#RefSeqFAQ.what_is_the_difference_between_1
  https://www.ncbi.nlm.nih.gov/refseq/about/)

---

### UniProt

There is lots of information available for proteins, such as sequence information, domains, expression, or 3D structure.
The aim of the Universal Protein Resource ([UniProt](https://www.uniprot.org/)) is to prove a comprehensive resource for proteins and their annotation.
UniProt contains three databases ({numref}`uniprot`):
* UniProt Knowledgebase (UniProtKB) -  see below
* UniProt Reference Clusters (UniRef) - clusters of protein sequences at 100%, 90%, and 50% identity
* UniProt Archive (UniParc) - non-redundant archive of publicly available protein sequences seen across different databases

:::{figure} images/Week1/uniprot.jpg
:alt: Overview of UniProt
:width: 100%
:name: uniprot

The information flow in Uniprot.
:::

**UniProtKB** is the central hub for functional information on proteins.
For each protein it contains the core data (such as sequence, name, description, taxonomy, citation) and as much annotation information as possible.
It contains many cross-references to other databases is generally a very good starting point to find information on a protein.

UniProtKB consists of two sections:
* Swiss-Prot - manually-annotated records with information extracted from literature and curated computational analysis
* TrEMBL - automatically annotated records that are not reviewed

UniProtKB is updated every 8 weeks. The current release has ~570,000 entries in Swiss-Prot and ~249 million entries in TrEMBL.

---

(sec_prosite)=
### Prosite

[Prosite](https://prosite.expasy.org/) is a secondary database of protein domains, families, and functional sites.
Some regions in protein families are more conserved than others because they are important for the structure or function of the protein.
Prosite contains motifs and profiles specific for many protein families or domains.
Searching motifs in new proteins can provide a first hint for protein function.

The current release of Prosite from 3 May 2023 contains 1311 patterns, 1363 profiles, and 1379 ProRule entries.

A Prosite **pattern** is typically 10 to 20 amino acids in length.
These short patterns are usually located in short well-conserved regions, such as  catalytic sites in enzymes or binding sites.
A pattern is represented as a regular expression, where amino acids are separated by hyphens and `x` denotes any letter.
Repetitions can also be given as the number of repetitions in brackets.
For example, `[AC]-x-V-x(4)-{ED}` matches sequences that contain the following amino acid sequence: Ala or Cys-any-Val-any-any-any-any-any but Glu or Asp.
Note that this representation is **qualitative**, a sequence either matches a pattern or it does not.

Patterns cannot deal with mismatches and are limited to exact matches to the pattern.
Thus, they are not well suited to identify distant homologs.
A Prosite **profile** is more general than a pattern and can also detect poorly conserved domains or families.
They characterize protein domains over their entire length and do not just model the conserved parts.
Profiles are estimated from multiple sequence alignments and we learn more about them in Week 2.
For now, it is important to know that profiles model matches, insertions, and deletions.
Importantly, profiles are **quantitative** representations, they will return a score how well the sequence fits to the profile.
A threshold can be applied to get high-scoring profiles for a sequence.
In contrast to patterns, a mismatch to a profile can be accepted if the rest of the sequence is highly similar to the profile.
Profiles are well suited to model structure properties of a domain.

Notably, profiles cover the structural relationships of domains, but they might also score a sequence highly that lacks important functional residues.
To include that information, **ProRule** contains additional information about Prosite profiles, such as the position of structurally or functionally important amino acids.
ProRule is used to guide curated annotation of UniProtKB/Swiss-Prot.

[comment]: # (source: https://academic.oup.com/bib/article/3/3/265/239712 https://academic.oup.com/bioinformatics/article/21/21/4060/226654
  example: https://prosite.expasy.org/rule/PRU01217)

### InterPro

The Integrated Resource of Protein Families, Domains and Sites ([InterPro](https://www.ebi.ac.uk/interpro/)) integrates 13 member databases (including [Prosite](sec_prosite) and [Pfam](sec_pfam)) into a comprehensive secondary database.
Additionally, it provides annotation from other tools, for example to annotate signal peptides and transmembrane regions.
It allows to identify functionally important domains and conserved sites in a sequence by simultaneously annotating it using the member databases.
Interpro can be used to find out to which protein family a sequence belongs to, or what its putative function is.
Additionally, one InterPro entry can integrate entries from the member databases, if they represent the same biological entity, reducing redundancy.
InterPro entries are also linked to [Gene Ontology](sec_go).
They are curated before being released.

InterPro is updated every 8 weeks. The current release from 10 May 2023 contains ~39,000 entries, which represent different types:

As an example, look at the [InterPro entry](https://www.ebi.ac.uk/interpro/entry/InterPro/IPR010945/) for the type 2 malate dehydrogenase protein family.
The entry has a name (malate dehydrogenase, type 2) and accession (IPR010945).
The contributing entries in member databases are shown on the right-hand side, with links to the individual member database entries.
A descriptive abstract explains what these proteins are and what their function is.
A set of GO terms is also provided, which describe the characteristics of the proteins matched by the entry.

You can get the InterPro annotation for a protein by running a new sequence search ({numref}`interpro-search`), or by by looking up its UniProt Accession ({numref}`interpro-browse`).

:::{figure} images/Week1/interpro-search.png
:alt: Searching interpro
:width: 100%
:name: interpro-search

Search fields on the InterPro home page, showing text search field (A) and the sequence search (B) options, including ‘Advanced options’, where you can limit your search to those member databases or sequence features of interest. Selecting the browse tab in the top menu (C) allows access to a browse search, (e.g., search for member database signature, InterPro entry type), see also {numref}`interpro-browse`. You can also search for a particular domain architecture (D).
:::

:::{figure} images/Week1/interpro-browse.png
:alt: Browsing interpro
:width: 100%
:name: interpro-browse

Browse the annotated proteins in Interpro and search for a UniProt accession.
See resulting entry in ({numref}`interpro-prot`).
:::

:::{figure} images/Week1/interpro-prot.png
:alt: A UniProt entry in Interpro
:width: 100%
:name: interpro-prot

The result page when looking up UniProt accession [A0A076FRI5](https://www.ebi.ac.uk/interpro/protein/UniProt/A0A076FRI5/) in InterPro.
You can see the family and domain annotation and on the right the accessions in InterPro and in the member databases.
You can click on each of these accessions to get to the entry information.
:::

You may have noticed a colored letter before each InterPro accession, e.g., F before IPR011835 or D before IPR001296 ({numref}`interpro-prot`).
These icons denote the different InterPro entry types:
* Family - a group of proteins sharing a common evolutionary origin, reflected by their related functions and similarities in sequence or structure.
* Superfamily - a large diverse family, usually with shared protein structure
* Domain - a distinct functional or structural unit in a protein, usually responsible for a particular function or interaction
* Site - a group of amino acids with certain characteristics that may be important for protein function, e.g., active sites or binding sites
* Repeat - typically a short amino acid sequence that is repeated within a protein

:::{figure} images/Week1/interpro-types.png
:alt: Interpro types
:width: 100%
:name: interpro-types

The icons for the different InterPro entries (homologous superfamily, family, domain, repeat or site).
:::

:::{seealso}
You can find more information on InterPro entry types with examples [here](https://www.ebi.ac.uk/training/online/courses/interpro-functional-and-structural-analysis/what-is-an-interpro-entry/interpro-entry-types/
).
:::

[comment]: # (source: https://www.ebi.ac.uk/training/online/courses/interpro-functional-and-structural-analysis/ https://academic.oup.com/nar/article/51/D1/D418/6814474)  

(sec_pfam)=
#### Pfam

Pfam is an important resource for protein domains.
In Pfam, domains are classified according to profiles that are modelled as Hidden Markov models (HMMs).
We will learn more on HMMs in Week 2.
Pfam is now integrated in InterPro.
Each Pfam domain can be represented with a logo, where the amino acids frequent at a particular position are represented as larger letters ({numref}`pfam-profile`).

:::{figure} images/Week1/pfam-profile.png
:alt: Pfam profile
:width: 100%
:name: pfam-profile

The Pfam logo for PF12924.
:::

---

(sec_go)=
### Gene Ontology

[Gene ontology](http://geneontology.org/) is a knowledgebase for the function of genes.
Each term is embedded into a hierarchy, where specific terms point to more generic terms ({numref}`go`).

:::{figure} images/Week1/go.png
:alt: Go hierarchy
:width: 100%
:name: go

An extract of the Gene Ontology hierarchy.
:::

---

## Reproducible research
